scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -279,7 +279,9 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
 
     override def next(): Token = {
       val (newSepRegions, newTokenRef) = nextToken(curr.pos, curr.nextPos, sepRegions)
-      prevPos = curr.pointPos
+      // don't update prevPos on artificially inserted tokens
+      if (curr.token.isNot[Indentation.Indent] && curr.token.isNot[Indentation.Outdent])
+        prevPos = curr.pointPos
```

Comment:
I think we can also remove `TokenRef.pointPos`.
I added it only to fix smth similar to this issue. For any token exept `Indent` and `Outdent` `pointPos` equals to `TokenRef.pos`

Reply:
When removing `pointPos` some tests seem to fail. we are still using prevPos set by Indent and Outdent, only on the next token we don't update it, so that we point to the previous pos before Indentation tokens.

Reply:
I've tried to run a test on the snippet from the issue and it seems it's seems this fix doesn't help.
```scala
object a:
   def foo =
      try foo
        catch
          case a =>
        finally bar
```

output:
```
=> Obtained
    """|Template :
       |   def foo =
       |     try foo
       |     catch
       |       case a =>
       |     finally bar
       |Self    @@def foo =
       |Defn.Def def foo =
       |     try foo
       |     catch
       |       case a =>
       |     finally bar
       |Term.Try try foo
       |     catch
       |       case a =>
       |     finally bar
       |Case case a =>
       |Term.Block => // <--- `=>` is printed 
       |""".stripMargin

```

Reply:
ok it seems that the fix is not working everywhere :/ Need to dig in more.

Reply:
Ok, another approach. Setting empty position on purpose.